### PR TITLE
Backlog 7168

### PR DESCRIPTION
**Pull request by denismashukov**
Added an additional operation and function to the unknown of expression, which is always FALSE

### CODE REVIEW

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. mvn -B -fn -DrunITs=true -f pdi-dataservice-client/pom.xml clean install && mvn -B -Dpentaho-coding-standards.version=1.0.4 -f pdi-dataservice-client/pom.xml site

Cleanup:
-------
mvn -B build-helper:remove-project-artifact

Changed files:
==============
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java

Unit test coverage change:
============
org.pentaho.di.core.jdbc.FoundClause:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.core.jdbc.ThinUtil:
 * Branch Change : **-82.95%**
 * Complexity Change : **-73.65%**
 * Instruction Change : **-90.95%**
 * Line Change : **-89.04%**
 * Method Change : **-93.10%**

org.pentaho.di.core.sql.IifFunction:
 * Branch Change : **-38.89%**
 * Complexity Change : **-33.33%**
 * Instruction Change : **-54.55%**
 * Line Change : **-57.45%**
 * Method Change : **-50.00%**

org.pentaho.di.core.sql.SQL:
 * Branch Change : **-80.00%**
 * Complexity Change : **-50.91%**
 * Instruction Change : **-76.48%**
 * Line Change : **-61.16%**
 * Method Change : **-45.71%**

org.pentaho.di.core.sql.SQLAggregation:
 * Complexity Change : **-80.00%**
 * Instruction Change : **-93.67%**
 * Line Change : **-100.00%**
 * Method Change : **-80.00%**

org.pentaho.di.core.sql.SQLCondition:
 * Branch Change : **-65.56%**
 * Complexity Change : **-56.72%**
 * Instruction Change : **-79.33%**
 * Line Change : **-77.96%**
 * Method Change : **-59.09%**

org.pentaho.di.core.sql.SQLField:
 * Branch Change : **-70.67%**
 * Complexity Change : **-49.32%**
 * Instruction Change : **-68.19%**
 * Line Change : **-62.42%**
 * Method Change : **-45.71%**

org.pentaho.di.core.sql.SQLFields:
 * Branch Change : **-79.63%**
 * Complexity Change : **-66.00%**
 * Instruction Change : **-90.71%**
 * Line Change : **-83.33%**
 * Method Change : **-73.91%**

org.pentaho.di.core.sql.SQLLimit:
 * Branch Change : **-70.00%**
 * Complexity Change : **-54.55%**
 * Instruction Change : **-92.68%**
 * Line Change : **-83.87%**
 * Method Change : **-66.67%**

org.pentaho.di.trans.dataservice.client.DataServiceClientPlugin:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.client.DataServiceLifecycleListener:
 * Branch Change : **-100.00%**
 * Complexity Change : **-90.91%**
 * Instruction Change : **-93.33%**
 * Line Change : **-92.59%**
 * Method Change : **-85.71%**

org.pentaho.di.trans.dataservice.jdbc.BaseResultSet:
 * Branch Change : **-78.57%**
 * Complexity Change : **-87.32%**
 * Instruction Change : **-91.40%**
 * Line Change : **-89.47%**
 * Method Change : **-91.89%**

org.pentaho.di.trans.dataservice.jdbc.BaseResultSet.new BaseResultSet.ValueRetriever() {...}:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.RemoteClient:
 * Branch Change : **-93.75%**
 * Complexity Change : **-76.47%**
 * Instruction Change : **-86.77%**
 * Line Change : **-80.33%**
 * Method Change : **-66.67%**

org.pentaho.di.trans.dataservice.jdbc.RowsResultSet:
 * Branch Change : **-100.00%**
 * Complexity Change : **-86.67%**
 * Instruction Change : **-87.30%**
 * Line Change : **-87.50%**
 * Method Change : **-84.62%**

org.pentaho.di.trans.dataservice.jdbc.ThinBase:
 * Branch Change : **-50.00%**
 * Complexity Change : **-85.71%**
 * Instruction Change : **-90.24%**
 * Line Change : **-90.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection:
 * Branch Change : **-58.33%**
 * Complexity Change : **-88.00%**
 * Instruction Change : **-95.56%**
 * Line Change : **-93.68%**
 * Method Change : **-94.20%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection.Builder:
 * Branch Change : **-92.86%**
 * Complexity Change : **-92.86%**
 * Instruction Change : **-88.39%**
 * Line Change : **-87.50%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection.new Predicate() {...}:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinDatabaseMetaData:
 * Branch Change : **-42.86%**
 * Complexity Change : **-67.89%**
 * Instruction Change : **-89.33%**
 * Line Change : **-83.89%**
 * Method Change : **-71.59%**

org.pentaho.di.trans.dataservice.jdbc.ThinDriver:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-89.55%**
 * Line Change : **-92.31%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinParameterMetaData:
 * Complexity Change : **-16.67%**
 * Instruction Change : **-22.73%**
 * Line Change : **-28.57%**
 * Method Change : **-16.67%**

org.pentaho.di.trans.dataservice.jdbc.ThinPreparedStatement:
 * Branch Change : **-67.92%**
 * Complexity Change : **-77.42%**
 * Instruction Change : **-80.49%**
 * Line Change : **-81.07%**
 * Method Change : **-90.62%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultFactory:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultHeader:
 * Complexity Change : **-57.14%**
 * Instruction Change : **-76.92%**
 * Line Change : **-78.57%**
 * Method Change : **-57.14%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultSet:
 * Branch Change : **-95.00%**
 * Complexity Change : **-96.00%**
 * Instruction Change : **-88.64%**
 * Line Change : **-94.23%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultSetMetaData:
 * Complexity Change : **-10.81%**
 * Instruction Change : **-15.82%**
 * Line Change : **-16.67%**
 * Method Change : **-18.18%**

org.pentaho.di.trans.dataservice.jdbc.ThinServiceInformation:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinStatement:
 * Branch Change : **-75.00%**
 * Complexity Change : **-80.77%**
 * Instruction Change : **-89.95%**
 * Line Change : **-92.06%**
 * Method Change : **-88.10%**

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. mvn -B -fn -DrunITs=true -f pdi-dataservice-client/pom.xml clean install && mvn -B -Dpentaho-coding-standards.version=1.0.4 -f pdi-dataservice-client/pom.xml site

Cleanup:
-------
mvn -B build-helper:remove-project-artifact

Changed files:
==============
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLConditionTest.java

Unit test coverage change:
============
org.pentaho.di.core.jdbc.FoundClause:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.core.jdbc.ThinUtil:
 * Branch Change : **-82.95%**
 * Complexity Change : **-73.65%**
 * Instruction Change : **-90.95%**
 * Line Change : **-89.04%**
 * Method Change : **-93.10%**

org.pentaho.di.core.sql.IifFunction:
 * Branch Change : **-38.89%**
 * Complexity Change : **-33.33%**
 * Instruction Change : **-54.55%**
 * Line Change : **-57.45%**
 * Method Change : **-50.00%**

org.pentaho.di.core.sql.SQL:
 * Branch Change : **-80.00%**
 * Complexity Change : **-50.91%**
 * Instruction Change : **-76.48%**
 * Line Change : **-61.16%**
 * Method Change : **-45.71%**

org.pentaho.di.core.sql.SQLAggregation:
 * Complexity Change : **-80.00%**
 * Instruction Change : **-93.67%**
 * Line Change : **-100.00%**
 * Method Change : **-80.00%**

org.pentaho.di.core.sql.SQLCondition:
 * Branch Change : **-65.56%**
 * Complexity Change : **-56.72%**
 * Instruction Change : **-79.33%**
 * Line Change : **-77.96%**
 * Method Change : **-59.09%**

org.pentaho.di.core.sql.SQLField:
 * Branch Change : **-70.67%**
 * Complexity Change : **-49.32%**
 * Instruction Change : **-68.19%**
 * Line Change : **-62.42%**
 * Method Change : **-45.71%**

org.pentaho.di.core.sql.SQLFields:
 * Branch Change : **-79.63%**
 * Complexity Change : **-66.00%**
 * Instruction Change : **-90.71%**
 * Line Change : **-83.33%**
 * Method Change : **-73.91%**

org.pentaho.di.core.sql.SQLLimit:
 * Branch Change : **-70.00%**
 * Complexity Change : **-54.55%**
 * Instruction Change : **-92.68%**
 * Line Change : **-83.87%**
 * Method Change : **-66.67%**

org.pentaho.di.trans.dataservice.client.DataServiceClientPlugin:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.client.DataServiceLifecycleListener:
 * Branch Change : **-100.00%**
 * Complexity Change : **-90.91%**
 * Instruction Change : **-93.33%**
 * Line Change : **-92.59%**
 * Method Change : **-85.71%**

org.pentaho.di.trans.dataservice.jdbc.BaseResultSet:
 * Branch Change : **-78.57%**
 * Complexity Change : **-87.32%**
 * Instruction Change : **-91.40%**
 * Line Change : **-89.47%**
 * Method Change : **-91.89%**

org.pentaho.di.trans.dataservice.jdbc.BaseResultSet.new BaseResultSet.ValueRetriever() {...}:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.RemoteClient:
 * Branch Change : **-93.75%**
 * Complexity Change : **-76.47%**
 * Instruction Change : **-86.77%**
 * Line Change : **-80.33%**
 * Method Change : **-66.67%**

org.pentaho.di.trans.dataservice.jdbc.RowsResultSet:
 * Branch Change : **-100.00%**
 * Complexity Change : **-86.67%**
 * Instruction Change : **-87.30%**
 * Line Change : **-87.50%**
 * Method Change : **-84.62%**

org.pentaho.di.trans.dataservice.jdbc.ThinBase:
 * Branch Change : **-50.00%**
 * Complexity Change : **-85.71%**
 * Instruction Change : **-90.24%**
 * Line Change : **-90.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection:
 * Branch Change : **-58.33%**
 * Complexity Change : **-88.00%**
 * Instruction Change : **-95.56%**
 * Line Change : **-93.68%**
 * Method Change : **-94.20%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection.Builder:
 * Branch Change : **-92.86%**
 * Complexity Change : **-92.86%**
 * Instruction Change : **-88.39%**
 * Line Change : **-87.50%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection.new Predicate() {...}:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinDatabaseMetaData:
 * Branch Change : **-42.86%**
 * Complexity Change : **-67.89%**
 * Instruction Change : **-89.33%**
 * Line Change : **-83.89%**
 * Method Change : **-71.59%**

org.pentaho.di.trans.dataservice.jdbc.ThinDriver:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-89.55%**
 * Line Change : **-92.31%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinParameterMetaData:
 * Complexity Change : **-16.67%**
 * Instruction Change : **-22.73%**
 * Line Change : **-28.57%**
 * Method Change : **-16.67%**

org.pentaho.di.trans.dataservice.jdbc.ThinPreparedStatement:
 * Branch Change : **-67.92%**
 * Complexity Change : **-77.42%**
 * Instruction Change : **-80.49%**
 * Line Change : **-81.07%**
 * Method Change : **-90.62%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultFactory:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultHeader:
 * Complexity Change : **-57.14%**
 * Instruction Change : **-76.92%**
 * Line Change : **-78.57%**
 * Method Change : **-57.14%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultSet:
 * Branch Change : **-95.00%**
 * Complexity Change : **-96.00%**
 * Instruction Change : **-88.64%**
 * Line Change : **-94.23%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultSetMetaData:
 * Complexity Change : **-10.81%**
 * Instruction Change : **-15.82%**
 * Line Change : **-16.67%**
 * Method Change : **-18.18%**

org.pentaho.di.trans.dataservice.jdbc.ThinServiceInformation:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinStatement:
 * Branch Change : **-75.00%**
 * Complexity Change : **-80.77%**
 * Instruction Change : **-89.95%**
 * Line Change : **-92.06%**
 * Method Change : **-88.10%**

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. mvn -B -fn -DrunITs=true -f pdi-dataservice-client/pom.xml clean install && mvn -B -Dpentaho-coding-standards.version=1.0.4 -f pdi-dataservice-client/pom.xml site

Cleanup:
-------
mvn -B build-helper:remove-project-artifact

Changed files:
==============
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLConditionTest.java

Unit test coverage change:
============
org.pentaho.di.core.jdbc.FoundClause:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.core.jdbc.ThinUtil:
 * Branch Change : **-82.95%**
 * Complexity Change : **-73.65%**
 * Instruction Change : **-90.95%**
 * Line Change : **-89.04%**
 * Method Change : **-93.10%**

org.pentaho.di.core.sql.IifFunction:
 * Branch Change : **-38.89%**
 * Complexity Change : **-33.33%**
 * Instruction Change : **-54.55%**
 * Line Change : **-57.45%**
 * Method Change : **-50.00%**

org.pentaho.di.core.sql.SQL:
 * Branch Change : **-80.00%**
 * Complexity Change : **-50.91%**
 * Instruction Change : **-76.48%**
 * Line Change : **-61.16%**
 * Method Change : **-45.71%**

org.pentaho.di.core.sql.SQLAggregation:
 * Complexity Change : **-80.00%**
 * Instruction Change : **-93.67%**
 * Line Change : **-100.00%**
 * Method Change : **-80.00%**

org.pentaho.di.core.sql.SQLCondition:
 * Branch Change : **-65.56%**
 * Complexity Change : **-56.72%**
 * Instruction Change : **-79.33%**
 * Line Change : **-77.96%**
 * Method Change : **-59.09%**

org.pentaho.di.core.sql.SQLField:
 * Branch Change : **-70.67%**
 * Complexity Change : **-49.32%**
 * Instruction Change : **-68.19%**
 * Line Change : **-62.42%**
 * Method Change : **-45.71%**

org.pentaho.di.core.sql.SQLFields:
 * Branch Change : **-79.63%**
 * Complexity Change : **-66.00%**
 * Instruction Change : **-90.71%**
 * Line Change : **-83.33%**
 * Method Change : **-73.91%**

org.pentaho.di.core.sql.SQLLimit:
 * Branch Change : **-70.00%**
 * Complexity Change : **-54.55%**
 * Instruction Change : **-92.68%**
 * Line Change : **-83.87%**
 * Method Change : **-66.67%**

org.pentaho.di.trans.dataservice.client.DataServiceClientPlugin:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.client.DataServiceLifecycleListener:
 * Branch Change : **-100.00%**
 * Complexity Change : **-90.91%**
 * Instruction Change : **-93.33%**
 * Line Change : **-92.59%**
 * Method Change : **-85.71%**

org.pentaho.di.trans.dataservice.jdbc.BaseResultSet:
 * Branch Change : **-78.57%**
 * Complexity Change : **-87.32%**
 * Instruction Change : **-91.40%**
 * Line Change : **-89.47%**
 * Method Change : **-91.89%**

org.pentaho.di.trans.dataservice.jdbc.BaseResultSet.new BaseResultSet.ValueRetriever() {...}:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.RemoteClient:
 * Branch Change : **-93.75%**
 * Complexity Change : **-76.47%**
 * Instruction Change : **-86.77%**
 * Line Change : **-80.33%**
 * Method Change : **-66.67%**

org.pentaho.di.trans.dataservice.jdbc.RowsResultSet:
 * Branch Change : **-100.00%**
 * Complexity Change : **-86.67%**
 * Instruction Change : **-87.30%**
 * Line Change : **-87.50%**
 * Method Change : **-84.62%**

org.pentaho.di.trans.dataservice.jdbc.ThinBase:
 * Branch Change : **-50.00%**
 * Complexity Change : **-85.71%**
 * Instruction Change : **-90.24%**
 * Line Change : **-90.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection:
 * Branch Change : **-58.33%**
 * Complexity Change : **-88.00%**
 * Instruction Change : **-95.56%**
 * Line Change : **-93.68%**
 * Method Change : **-94.20%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection.Builder:
 * Branch Change : **-92.86%**
 * Complexity Change : **-92.86%**
 * Instruction Change : **-88.39%**
 * Line Change : **-87.50%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinConnection.new Predicate() {...}:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinDatabaseMetaData:
 * Branch Change : **-42.86%**
 * Complexity Change : **-67.89%**
 * Instruction Change : **-89.33%**
 * Line Change : **-83.89%**
 * Method Change : **-71.59%**

org.pentaho.di.trans.dataservice.jdbc.ThinDriver:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-89.55%**
 * Line Change : **-92.31%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinParameterMetaData:
 * Complexity Change : **-16.67%**
 * Instruction Change : **-22.73%**
 * Line Change : **-28.57%**
 * Method Change : **-16.67%**

org.pentaho.di.trans.dataservice.jdbc.ThinPreparedStatement:
 * Branch Change : **-67.92%**
 * Complexity Change : **-77.42%**
 * Instruction Change : **-80.49%**
 * Line Change : **-81.07%**
 * Method Change : **-90.62%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultFactory:
 * Branch Change : **-100.00%**
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultHeader:
 * Complexity Change : **-57.14%**
 * Instruction Change : **-76.92%**
 * Line Change : **-78.57%**
 * Method Change : **-57.14%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultSet:
 * Branch Change : **-95.00%**
 * Complexity Change : **-96.00%**
 * Instruction Change : **-88.64%**
 * Line Change : **-94.23%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinResultSetMetaData:
 * Complexity Change : **-10.81%**
 * Instruction Change : **-15.82%**
 * Line Change : **-16.67%**
 * Method Change : **-18.18%**

org.pentaho.di.trans.dataservice.jdbc.ThinServiceInformation:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.trans.dataservice.jdbc.ThinStatement:
 * Branch Change : **-75.00%**
 * Complexity Change : **-80.77%**
 * Instruction Change : **-89.95%**
 * Line Change : **-92.06%**
 * Method Change : **-88.10%**

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. mvn -B -fn -DrunITs=true -f pdi-dataservice-client/pom.xml clean install && mvn -B -Dpentaho-coding-standards.version=1.0.4 -f pdi-dataservice-client/pom.xml site

Cleanup:
-------
mvn -B build-helper:remove-project-artifact

Changed files:
==============
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLConditionTest.java

Newly Broken Tests:
======================
org.pentaho.di.core.sql.SQLConditionTest.testCondition31:
-----------------------------------------------
junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError
	at org.pentaho.di.core.sql.SQLConditionTest.testCondition31(SQLConditionTest.java:933)

```

Unit test coverage change:
============
org.pentaho.di.core.sql.SQLCondition:
 * Branch Change :  + .75%
 * Complexity Change :  + .64%
 * Instruction Change :  + .75%
 * Line Change :  + .35%

Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards
File /home/buildguy/project/pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java

1. Message: '(' is not followed by whitespace. at Line: 283 Column: 40
2. Message: '(' is not followed by whitespace. at Line: 283 Column: 52
3. Message: ')' is not preceded with whitespace. at Line: 283 Column: 53
4. Message: ')' is not preceded with whitespace. at Line: 283 Column: 54
5. Message: '(' is not followed by whitespace. at Line: 351 Column: 28
6. Message: '(' is not followed by whitespace. at Line: 353 Column: 28
7. Message: '(' is not followed by whitespace. at Line: 401 Column: 75
8. Message: ')' is not preceded with whitespace. at Line: 401 Column: 83

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. mvn -B -fn -DrunITs=true -f pdi-dataservice-client/pom.xml clean install && mvn -B -Dpentaho-coding-standards.version=1.0.4 -f pdi-dataservice-client/pom.xml site

Cleanup:
-------
mvn -B build-helper:remove-project-artifact

Changed files:
==============
  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLConditionTest.java

Newly Broken Tests:
======================
org.pentaho.di.core.sql.SQLConditionTest.testCondition31:
-----------------------------------------------
junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError
	at org.pentaho.di.core.sql.SQLConditionTest.testCondition31(SQLConditionTest.java:933)

```

Unit test coverage change:
============
org.pentaho.di.core.sql.SQLCondition:
 * Branch Change :  + .75%
 * Complexity Change :  + .64%
 * Instruction Change :  + .67%
 * Line Change :  + .35%

**Comment by brosander** [comment]
# Build Commands:

## Build:
1. mvn -B -fn -DrunITs=true -f pdi-dataservice-client/pom.xml clean install && mvn -B -Dpentaho-coding-standards.version=1.0.4 -f pdi-dataservice-client/pom.xml site

## Cleanup:

mvn -B build-helper:remove-project-artifact

# Changed files:

  pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java
  pdi-dataservice-client/src/test/java/org/pentaho/di/core/sql/SQLConditionTest.java

# Unit test coverage change:

org.pentaho.di.core.sql.SQLCondition:
- Branch Change :  + .75%
- Complexity Change :  + .64%
- Instruction Change :  + .67%
- Line Change :  + .35%

File: pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java [hunk 1]
```diff
@@ -372,28 +374,35 @@ private Condition parseAtomicCondition( String clause ) throws KettleSQLExceptio
...
-            strings.add( op );
-            strings.add( right );
-            return strings;
+
+            if ( Pattern.matches( "^NULL$", right.toUpperCase() ) ) {
+              op = Condition.functions[Condition.FUNC_TRUE];
+              negation = true;
+              right = "";
+            }
+
```
Comment (mkambol) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java]: @denis8703 It seems weird to be constructing a Condition in this method, only to then return a string array which later needs to be interpreted and have a Condition constructed.  Can the check for ^NULL$ in the right expression be done in parseAtomicCondition?

**Comment by mkambol** [comment]
@denis8703 I tested out your change, and it actually results in all rows being returned when you query a dataservice.  It looks like my suggestion for negating a TRUE is not quite so simple-- the Condition.evaluate checks for true and then returns immediately.  So we'll need to modify Condition to look at the negate value when the function is TRUE (or find some other way to represent FALSE)

https://github.com/pentaho/pentaho-kettle/blob/master/core/src/org/pentaho/di/core/Condition.java#L377

**Comment by denismashukov** [comment]
@mkambol Matt I see, I made the changes for Condition.evaluate 
https://github.com/pentaho/pentaho-kettle/pull/2450/files

File: pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java (hunk 1 above)
Comment (denismashukov) [inline pdi-dataservice-client/src/main/java/org/pentaho/di/core/sql/SQLCondition.java]: changed

**Comment by denismashukov** [comment]
maybe it would be easier to add a FUNC_FALSE, without complication

**Pull request merged**